app/src/main/java/com/concordia/mcga/fragments/IndoorMapFragment.java

```diff
@@ -71,53 +76,88 @@ public void onPageFinished(WebView view, String url) {
         //Floor Select Spinner
         floorButtonContainer = (LinearLayout) view.findViewById(R.id.floorButtonContainer);
 
-        //Indoor POI Stack
-        indoorPoiStack = new ArrayList<IndoorPOI>();
+        //Progress Bar
+        pathProgressBar = (ProgressBar) view.findViewById(R.id.pathProgressBar);
+        pathProgressBar.setVisibility(View.GONE);
+
+        //Init Attributes
+        indoorPoiStack = new ArrayList<>();
+        currentPathTiles = new HashMap<>();
+        floorsLoaded = new HashMap<>();
+
+        Campus.populateCampusesWithBuildings();
 
         return view;
     }
 
     public void generatePath(final IndoorPOI start, final IndoorPOI dest) {
-        leafletView.post(new Runnable() {
-            @Override
-            public void run() {
-                Building hall = new Building(new LatLng(0,0), "Hall", "H", new MarkerOptions());
-                Floor H4 = hall.getFloorMap(4);
-                SingleMapPathFinder pf = new SingleMapPathFinder(H4.getMap());
-
-                ArrayList<IndoorMapTile> pathTilesJunctions = null;
+        Thread generatePathThread = null;
 
-                try {
-                    pathTilesJunctions = (ArrayList<IndoorMapTile>) pf.shortestPathJunctions(start.getTile(), dest.getTile());
+        if (!pathGenerating) {
+            generatePathThread = new Thread(new GeneratePath(start, dest));
+        }
 
-                    Iterator<IndoorMapTile> it2 = pathTilesJunctions.iterator();
-                    while (it2.hasNext()) {
-                        IndoorMapTile pft2 = it2.next();
-                        Log.d("JCT: ", pft2.toString());
-                    }
+        generatePathThread.start();
```

Comment:
this will throw an exception if the user tries to generate a new path while the pathGenerating is true